Improve error logging in case of errors

**Description of the issue**:

This is the highest level of debugging verbosity I could achieve:
mvn -Dhttps.proxyHost=proxy -Dhttps.proxyPort=3128 compile jib:build -DjibSerialize=true -Djava.util.logging.config.file=logger.properties -X

logging.properties:
```
handlers = java.util.logging.ConsoleHandler
java.util.logging.ConsoleHandler.level=ALL

# CONFIG hides authentication data
# ALL includes authentication data
com.google.api.client.http.level=CONFIG
```

Even at this level I was not able to figure out the reason why the push into the repository did not succeeded. (See log.)

The actual problem was that in front of nexus3 there is was an nginx proxy which did not allow the request through. (Docker did return the cause of the problem:) 
```
5e93ace71c0b: Pushed 
ff3a3ba9afd4: Pushed 
8eefbd08e143: Pushing [==================================================>] 40.84 MB/40.84 MB
5b0bbc8b30cc: Pushing [==================================================>] 99.88 MB
6189abe095d5: Pushing [==================================================>] 1.966 MB
c5204564c844: Pushing [==================================================>] 16.82 MB/16.82 MB
error parsing HTTP 413 response body: invalid character '<' looking for beginning of value: "<html>\r\n<head><title>413 Request Entity Too Large</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>413 Request Entity Too Large</h1></center>\r\n<hr><center>nginx/1.15.3</center>\r\n</body>\r\n</html>\r\n"
```

**Expected behavior**:

At least the http status or preferable the full HTTP request should be visible on the highest level of logging set. 

**Steps to reproduce**: <!-- Please provide a minimal and precise series of steps -->

Push anything greater than 50mb into a nexus3 using ingress in front.

**Environment**: 
Linux, Maven 3.6

**Log output**: 
```
[DEBUG] TIMED   Building container configuration : 2.0 ms
[DEBUG] TIMING  Pushing container configuration
[DEBUG] TIMING  Pushing BLOB digest: sha256:ebd874fd0c121bdab6f200b0c346c3d340a7664e23d8431ed04b261f06f08e5b, size: 1443
[DEBUG] BLOB : digest: sha256:ebd874fd0c121bdab6f200b0c346c3d340a7664e23d8431ed04b261f06f08e5b, size: 1443 already exists on registry
[DEBUG] TIMED   Pushing BLOB digest: sha256:ebd874fd0c121bdab6f200b0c346c3d340a7664e23d8431ed04b261f06f08e5b, size: 1443 : 157.0 ms
[DEBUG] TIMED   Pushing container configuration : 198.0 ms
[DEBUG] TIMING  Setting up to push layers
[DEBUG] TIMING  Pushing BLOB digest: sha256:0dc77d0dd653ff05c520685f7d93b62e80c0b6b7adaf5d1feba731e182654ad1, size: 35812034
[DEBUG] TIMING  pushBlob
[DEBUG]         TIMING  pushBlob POST sha256:0dc77d0dd653ff05c520685f7d93b62e80c0b6b7adaf5d1feba731e182654ad1
[DEBUG]         TIMED   pushBlob PATCH sha256:0dc77d0dd653ff05c520685f7d93b62e80c0b6b7adaf5d1feba731e182654ad1 : 305.0 ms
[DEBUG]         TIMED   pushBlob POST sha256:0dc77d0dd653ff05c520685f7d93b62e80c0b6b7adaf5d1feba731e182654ad1 : 796.0 ms
[DEBUG] TIMED   pushBlob : 1102.0 ms
[DEBUG] TIMED   Pushing BLOB digest: sha256:0dc77d0dd653ff05c520685f7d93b62e80c0b6b7adaf5d1feba731e182654ad1, size: 35812034 : 1217.0 ms
[DEBUG] TIMING  Pushing BLOB digest: sha256:e342ef7618561dbde6eebfb41eca8030994754bc4ef57bf393a5d03b34338ea2, size: 1058
[DEBUG] BLOB : digest: sha256:e342ef7618561dbde6eebfb41eca8030994754bc4ef57bf393a5d03b34338ea2, size: 1058 already exists on registry
[DEBUG] TIMED   Pushing BLOB digest: sha256:e342ef7618561dbde6eebfb41eca8030994754bc4ef57bf393a5d03b34338ea2, size: 1058 : 163.0 ms
[DEBUG] TIMING  Pushing BLOB digest: sha256:5815c1384845c5a2f15bf64737d711703528ce63cf41fecd144ed835c69361dd, size: 63863
[DEBUG] BLOB : digest: sha256:5815c1384845c5a2f15bf64737d711703528ce63cf41fecd144ed835c69361dd, size: 63863 already exists on registry
[DEBUG] TIMED   Pushing BLOB digest: sha256:5815c1384845c5a2f15bf64737d711703528ce63cf41fecd144ed835c69361dd, size: 63863 : 116.0 ms
[DEBUG] TIMED   Setting up to push layers : 1497.0 ms
[INFO] Finalizing...
[DEBUG] TIMED   Building and pushing image : 5903.0 ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.832 s
[INFO] Finished at: 2018-12-07T12:29:25+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.10.1:build (default-cli) on project Netflex: Build image failed: Connection has been shutdown: javax.net.ssl.SSLException: java.net.SocketException: Broken pipe (Write failed) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.10.1:build (default-cli) on project Netflex: Build image failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Build image failed
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:152)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: javax.net.ssl.SSLException: Connection has been shutdown: javax.net.ssl.SSLException: java.net.SocketException: Broken pipe (Write failed)
    at sun.security.ssl.SSLSocketImpl.checkEOF (SSLSocketImpl.java:1533)
    at sun.security.ssl.SSLSocketImpl.checkWrite (SSLSocketImpl.java:1545)
    at sun.security.ssl.AppOutputStream.write (AppOutputStream.java:71)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.flushBuffer (AbstractSessionOutputBuffer.java:106)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.flush (AbstractSessionOutputBuffer.java:113)
    at org.apache.http.impl.AbstractHttpClientConnection.doFlush (AbstractHttpClientConnection.java:260)
    at org.apache.http.impl.SocketHttpClientConnection.close (SocketHttpClientConnection.java:248)
    at org.apache.http.impl.conn.DefaultClientConnection.close (DefaultClientConnection.java:154)
    at org.apache.http.impl.conn.AbstractPooledConnAdapter.close (AbstractPooledConnAdapter.java:148)
    at org.apache.http.protocol.HttpRequestExecutor.execute (HttpRequestExecutor.java:130)
    at org.apache.http.impl.client.DefaultRequestDirector.execute (DefaultRequestDirector.java:483)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:641)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:576)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:554)
    at com.google.api.client.http.apache.ApacheHttpRequest.execute (ApacheHttpRequest.java:65)
    at com.google.api.client.http.HttpRequest.execute (HttpRequest.java:981)
    at com.google.cloud.tools.jib.http.Connection.send (Connection.java:161)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:232)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling (RegistryEndpointCaller.java:152)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:142)
    at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint (RegistryClient.java:339)
    at com.google.cloud.tools.jib.registry.RegistryClient.pushBlob (RegistryClient.java:297)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:88)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:36)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.<init> (PushBlobStep.java:60)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.makePushBlobStep (PushLayersStep.java:94)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.lambda$call$0 (PushLayersStep.java:78)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:78)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:32)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.<init> (PushLayersStep.java:58)
    at com.google.cloud.tools.jib.builder.steps.StepsRunner.runPushBaseImageLayersStep (StepsRunner.java:107)
    at com.google.cloud.tools.jib.builder.BuildSteps.lambda$forBuildToDockerRegistry$0 (BuildSteps.java:63)
    at com.google.cloud.tools.jib.builder.BuildSteps.run (BuildSteps.java:150)
    at com.google.cloud.tools.jib.api.JibContainerBuilder.containerize (JibContainerBuilder.java:421)
    at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build (BuildStepsRunner.java:212)
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: javax.net.ssl.SSLException: java.net.SocketException: Broken pipe (Write failed)
    at sun.security.ssl.Alerts.getSSLException (Alerts.java:208)
    at sun.security.ssl.SSLSocketImpl.fatal (SSLSocketImpl.java:1946)
    at sun.security.ssl.SSLSocketImpl.fatal (SSLSocketImpl.java:1903)
    at sun.security.ssl.SSLSocketImpl.handleException (SSLSocketImpl.java:1867)
    at sun.security.ssl.SSLSocketImpl.handleException (SSLSocketImpl.java:1812)
    at sun.security.ssl.AppOutputStream.write (AppOutputStream.java:128)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.flushBuffer (AbstractSessionOutputBuffer.java:106)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.write (AbstractSessionOutputBuffer.java:126)
    at org.apache.http.impl.io.ChunkedOutputStream.flushCacheWithAppend (ChunkedOutputStream.java:112)
    at org.apache.http.impl.io.ChunkedOutputStream.write (ChunkedOutputStream.java:164)
    at java.security.DigestOutputStream.write (DigestOutputStream.java:145)
    at com.google.cloud.tools.jib.hash.CountingDigestOutputStream.write (CountingDigestOutputStream.java:83)
    at com.google.common.io.ByteStreams.copy (ByteStreams.java:110)
    at com.google.cloud.tools.jib.blob.BlobDescriptor.fromPipe (BlobDescriptor.java:47)
    at com.google.cloud.tools.jib.blob.FileBlob.writeTo (FileBlob.java:38)
    at com.google.cloud.tools.jib.http.BlobHttpContent.writeTo (BlobHttpContent.java:53)
    at com.google.api.client.http.apache.ContentEntity.writeTo (ContentEntity.java:62)
    at org.apache.http.impl.entity.EntitySerializer.serialize (EntitySerializer.java:127)
    at org.apache.http.impl.AbstractHttpClientConnection.sendRequestEntity (AbstractHttpClientConnection.java:253)
    at org.apache.http.impl.conn.AbstractClientConnAdapter.sendRequestEntity (AbstractClientConnAdapter.java:218)
    at org.apache.http.protocol.HttpRequestExecutor.doSendRequest (HttpRequestExecutor.java:249)
    at org.apache.http.protocol.HttpRequestExecutor.execute (HttpRequestExecutor.java:124)
    at org.apache.http.impl.client.DefaultRequestDirector.execute (DefaultRequestDirector.java:483)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:641)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:576)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:554)
    at com.google.api.client.http.apache.ApacheHttpRequest.execute (ApacheHttpRequest.java:65)
    at com.google.api.client.http.HttpRequest.execute (HttpRequest.java:981)
    at com.google.cloud.tools.jib.http.Connection.send (Connection.java:161)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:232)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling (RegistryEndpointCaller.java:152)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:142)
    at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint (RegistryClient.java:339)
    at com.google.cloud.tools.jib.registry.RegistryClient.pushBlob (RegistryClient.java:297)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:88)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:36)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.<init> (PushBlobStep.java:60)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.makePushBlobStep (PushLayersStep.java:94)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.lambda$call$0 (PushLayersStep.java:78)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:78)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:32)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.<init> (PushLayersStep.java:58)
    at com.google.cloud.tools.jib.builder.steps.StepsRunner.runPushBaseImageLayersStep (StepsRunner.java:107)
    at com.google.cloud.tools.jib.builder.BuildSteps.lambda$forBuildToDockerRegistry$0 (BuildSteps.java:63)
    at com.google.cloud.tools.jib.builder.BuildSteps.run (BuildSteps.java:150)
    at com.google.cloud.tools.jib.api.JibContainerBuilder.containerize (JibContainerBuilder.java:421)
    at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build (BuildStepsRunner.java:212)
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.net.SocketException: Broken pipe (Write failed)
    at java.net.SocketOutputStream.socketWrite0 (Native Method)
    at java.net.SocketOutputStream.socketWrite (SocketOutputStream.java:111)
    at java.net.SocketOutputStream.write (SocketOutputStream.java:155)
    at sun.security.ssl.OutputRecord.writeBuffer (OutputRecord.java:431)
    at sun.security.ssl.OutputRecord.write (OutputRecord.java:417)
    at sun.security.ssl.SSLSocketImpl.writeRecordInternal (SSLSocketImpl.java:879)
    at sun.security.ssl.SSLSocketImpl.writeRecord (SSLSocketImpl.java:850)
    at sun.security.ssl.AppOutputStream.write (AppOutputStream.java:123)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.flushBuffer (AbstractSessionOutputBuffer.java:106)
    at org.apache.http.impl.io.AbstractSessionOutputBuffer.write (AbstractSessionOutputBuffer.java:126)
    at org.apache.http.impl.io.ChunkedOutputStream.flushCacheWithAppend (ChunkedOutputStream.java:112)
    at org.apache.http.impl.io.ChunkedOutputStream.write (ChunkedOutputStream.java:164)
    at java.security.DigestOutputStream.write (DigestOutputStream.java:145)
    at com.google.cloud.tools.jib.hash.CountingDigestOutputStream.write (CountingDigestOutputStream.java:83)
    at com.google.common.io.ByteStreams.copy (ByteStreams.java:110)
    at com.google.cloud.tools.jib.blob.BlobDescriptor.fromPipe (BlobDescriptor.java:47)
    at com.google.cloud.tools.jib.blob.FileBlob.writeTo (FileBlob.java:38)
    at com.google.cloud.tools.jib.http.BlobHttpContent.writeTo (BlobHttpContent.java:53)
    at com.google.api.client.http.apache.ContentEntity.writeTo (ContentEntity.java:62)
    at org.apache.http.impl.entity.EntitySerializer.serialize (EntitySerializer.java:127)
    at org.apache.http.impl.AbstractHttpClientConnection.sendRequestEntity (AbstractHttpClientConnection.java:253)
    at org.apache.http.impl.conn.AbstractClientConnAdapter.sendRequestEntity (AbstractClientConnAdapter.java:218)
    at org.apache.http.protocol.HttpRequestExecutor.doSendRequest (HttpRequestExecutor.java:249)
    at org.apache.http.protocol.HttpRequestExecutor.execute (HttpRequestExecutor.java:124)
    at org.apache.http.impl.client.DefaultRequestDirector.execute (DefaultRequestDirector.java:483)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:641)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:576)
    at org.apache.http.impl.client.AbstractHttpClient.execute (AbstractHttpClient.java:554)
    at com.google.api.client.http.apache.ApacheHttpRequest.execute (ApacheHttpRequest.java:65)
    at com.google.api.client.http.HttpRequest.execute (HttpRequest.java:981)
    at com.google.cloud.tools.jib.http.Connection.send (Connection.java:161)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:232)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling (RegistryEndpointCaller.java:152)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:142)
    at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint (RegistryClient.java:339)
    at com.google.cloud.tools.jib.registry.RegistryClient.pushBlob (RegistryClient.java:297)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:88)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:36)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.<init> (PushBlobStep.java:60)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.makePushBlobStep (PushLayersStep.java:94)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.lambda$call$0 (PushLayersStep.java:78)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:78)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.call (PushLayersStep.java:32)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:258)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureInterruptibleTask.execute (CombinedFuture.java:112)
    at com.google.common.util.concurrent.CombinedFuture$CombinedFutureRunningState.handleAllCompleted (CombinedFuture.java:75)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.processCompleted (AggregateFuture.java:261)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.decrementCountAndMaybeComplete (AggregateFuture.java:248)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$300 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture$RunningState$1.run (AggregateFuture.java:151)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute (MoreExecutors.java:397)
    at com.google.common.util.concurrent.AbstractFuture.executeListener (AbstractFuture.java:1016)
    at com.google.common.util.concurrent.AbstractFuture.addListener (AbstractFuture.java:672)
    at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener (AbstractFuture.java:107)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.init (AggregateFuture.java:144)
    at com.google.common.util.concurrent.AggregateFuture$RunningState.access$100 (AggregateFuture.java:94)
    at com.google.common.util.concurrent.AggregateFuture.init (AggregateFuture.java:91)
    at com.google.common.util.concurrent.CombinedFuture.<init> (CombinedFuture.java:52)
    at com.google.common.util.concurrent.Futures$FutureCombiner.call (Futures.java:993)
    at com.google.cloud.tools.jib.builder.steps.PushLayersStep.<init> (PushLayersStep.java:58)
    at com.google.cloud.tools.jib.builder.steps.StepsRunner.runPushBaseImageLayersStep (StepsRunner.java:107)
    at com.google.cloud.tools.jib.builder.BuildSteps.lambda$forBuildToDockerRegistry$0 (BuildSteps.java:63)
    at com.google.cloud.tools.jib.builder.BuildSteps.run (BuildSteps.java:150)
    at com.google.cloud.tools.jib.api.JibContainerBuilder.containerize (JibContainerBuilder.java:421)
    at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build (BuildStepsRunner.java:212)
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
@@ -1,0 +1,29 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.registry;
+
+/** Thrown when the registry shut down the connection. */
+public class RegistryBrokenPipeException extends RegistryException {
+
+  RegistryBrokenPipeException(Throwable cause) {
+    super(
+        "I/O error due to broken pipe: the server shut down the connection. "
+            + "Check the server log if possible. This could also be a proxy issue. For example,"
+            + "a proxy may prevent sending bytes that is too large.",
+        cause);
+  }
+}

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
@@ -23,7 +23,7 @@ class RegistryBrokenPipeException extends RegistryException {
     super(
         "I/O error due to broken pipe: the server shut down the connection. "
             + "Check the server log if possible. This could also be a proxy issue. For example,"
-            + "a proxy may prevent sending bytes that is too large.",
+            + "a proxy may prevent sending packets that are too large.",
         cause);
   }
 }

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryBrokenPipeException.java
@@ -17,7 +17,7 @@
 package com.google.cloud.tools.jib.registry;
 
 /** Thrown when the registry shut down the connection. */
-public class RegistryBrokenPipeException extends RegistryException {
+class RegistryBrokenPipeException extends RegistryException {
 
   RegistryBrokenPipeException(Throwable cause) {
     super(

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
@@ -63,10 +63,16 @@ class RegistryEndpointCaller<T> {
   // https://github.com/GoogleContainerTools/jib/issues/1316
   @VisibleForTesting
   static boolean isBrokenPipe(IOException original) {
-    for (Throwable exception = original; exception != null; exception = exception.getCause()) {
+    Throwable exception = original;
+    while (exception != null) {
       String message = exception.getMessage();
       if (message != null && message.toLowerCase(Locale.US).contains("broken pipe")) {
         return true;
+      }
+
+      exception = exception.getCause();
+      if (exception == original) { // just in case if there's a circular chain
+        return false;
       }
     }
     return false;

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
@@ -33,8 +33,10 @@ import com.google.common.annotations.VisibleForTesting;
 import java.io.IOException;
 import java.net.ConnectException;
 import java.net.MalformedURLException;
+import java.net.SocketException;
 import java.net.URL;
 import java.security.GeneralSecurityException;
+import java.util.Locale;
 import java.util.function.Function;
 import javax.annotation.Nullable;
 import javax.net.ssl.SSLException;
@@ -57,6 +59,18 @@ class RegistryEndpointCaller<T> {
 
   private static boolean isHttpsProtocol(URL url) {
     return "https".equals(url.getProtocol());
+  }
+
+  // https://github.com/GoogleContainerTools/jib/issues/1316
+  @VisibleForTesting
+  static boolean isBrokenPipe(IOException original) {
+    for (Throwable exception = original; exception != null; exception = exception.getCause()) {
+      String message = exception.getMessage();
+      if (message != null && message.toLowerCase(Locale.US).contains("broken pipe")) {
+        return true;
+      }
+    }
+    return false;
   }
 
   private final EventDispatcher eventDispatcher;
@@ -232,6 +246,12 @@ class RegistryEndpointCaller<T> {
           connection.send(registryEndpointProvider.getHttpMethod(), requestBuilder.build());
 
       return registryEndpointProvider.handleResponse(response);
+
+    } catch (SSLException | SocketException ex) {
+      if (isBrokenPipe(ex)) {
+        throw new RegistryBrokenPipeException(ex);
+      }
+      throw ex;
 
     } catch (HttpResponseException ex) {
       // First, see if the endpoint provider handles an exception as an expected response.

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java
@@ -33,7 +33,6 @@ import com.google.common.annotations.VisibleForTesting;
 import java.io.IOException;
 import java.net.ConnectException;
 import java.net.MalformedURLException;
-import java.net.SocketException;
 import java.net.URL;
 import java.security.GeneralSecurityException;
 import java.util.Locale;
@@ -247,12 +246,6 @@ class RegistryEndpointCaller<T> {
 
       return registryEndpointProvider.handleResponse(response);
 
-    } catch (SSLException | SocketException ex) {
-      if (isBrokenPipe(ex)) {
-        throw new RegistryBrokenPipeException(ex);
-      }
-      throw ex;
-
     } catch (HttpResponseException ex) {
       // First, see if the endpoint provider handles an exception as an expected response.
       try {
@@ -312,6 +305,12 @@ class RegistryEndpointCaller<T> {
       }
     } catch (NoHttpResponseException ex) {
       throw new RegistryNoResponseException(ex);
+
+    } catch (IOException ex) {
+      if (isBrokenPipe(ex)) {
+        throw new RegistryBrokenPipeException(ex);
+      }
+      throw ex;
     }
   }
 }

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java
@@ -22,7 +22,7 @@ import javax.annotation.Nullable;
  * Thrown when an HTTP request to a registry endpoint failed with errors as defined in {@link
  * ErrorCodes}.
  */
-public class RegistryErrorException extends RegistryException {
+class RegistryErrorException extends RegistryException {
 
   RegistryErrorException(String message, @Nullable Throwable cause) {
     super(message, cause);

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryNoResponseException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryNoResponseException.java
@@ -17,7 +17,7 @@
 package com.google.cloud.tools.jib.registry;
 
 /** Thrown when a registry did not respond. */
-public class RegistryNoResponseException extends RegistryException {
+class RegistryNoResponseException extends RegistryException {
 
   RegistryNoResponseException(Throwable cause) {
     super(cause);

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/UnexpectedBlobDigestException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/UnexpectedBlobDigestException.java
@@ -17,7 +17,7 @@
 package com.google.cloud.tools.jib.registry;
 
 /** Thrown when a pulled BLOB did not have the same digest as requested. */
-public class UnexpectedBlobDigestException extends RegistryException {
+class UnexpectedBlobDigestException extends RegistryException {
 
   UnexpectedBlobDigestException(String message) {
     super(message);

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryEndpointCallerTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryEndpointCallerTest.java
@@ -32,11 +32,13 @@ import com.google.cloud.tools.jib.http.Response;
 import java.io.IOException;
 import java.net.ConnectException;
 import java.net.MalformedURLException;
+import java.net.SocketException;
 import java.net.URL;
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
 import javax.annotation.Nullable;
+import javax.net.ssl.SSLException;
 import javax.net.ssl.SSLPeerUnverifiedException;
 import org.apache.http.NoHttpResponseException;
 import org.hamcrest.CoreMatchers;
@@ -483,6 +485,26 @@ public class RegistryEndpointCallerTest {
     secureEndpointCaller.call();
 
     Assert.assertEquals(Integer.valueOf(7593), mockConnection.getRequestedHttpTimeout());
+  }
+
+  @Test
+  public void testIsBrokenPipe_notBrokenPipe() {
+    Assert.assertFalse(RegistryEndpointCaller.isBrokenPipe(new IOException()));
+    Assert.assertFalse(RegistryEndpointCaller.isBrokenPipe(new SocketException()));
+    Assert.assertFalse(RegistryEndpointCaller.isBrokenPipe(new SSLException("mock")));
+  }
+
+  @Test
+  public void testIsBrokenPipe_brokenPipe() {
+    Assert.assertTrue(RegistryEndpointCaller.isBrokenPipe(new IOException("cool broken pipe !")));
+    Assert.assertTrue(RegistryEndpointCaller.isBrokenPipe(new SocketException("BROKEN PIPE")));
+    Assert.assertTrue(RegistryEndpointCaller.isBrokenPipe(new SSLException("calm BrOkEn PiPe")));
+  }
+
+  @Test
+  public void testIsBrokenPipe_nestedBrokenPipe() {
+    IOException exception = new IOException(new SSLException(new SocketException("Broken pipe")));
+    Assert.assertTrue(RegistryEndpointCaller.isBrokenPipe(exception));
   }
 
   /**

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryEndpointCallerTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryEndpointCallerTest.java
@@ -507,6 +507,14 @@ public class RegistryEndpointCallerTest {
     Assert.assertTrue(RegistryEndpointCaller.isBrokenPipe(exception));
   }
 
+  @Test
+  public void testIsBrokenPipe_terminatesWhenCauseIsOriginal() {
+    IOException exception = Mockito.mock(IOException.class);
+    Mockito.when(exception.getCause()).thenReturn(exception);
+
+    Assert.assertFalse(RegistryEndpointCaller.isBrokenPipe(exception));
+  }
+
   /**
    * Verifies that a response with {@code httpStatusCode} throws {@link
    * RegistryUnauthorizedException}.

